model_bakery/baker.py

```diff
@@ -57,27 +58,58 @@ def _valid_quantity(quantity: Optional[Union[str, int]]) -> bool:
     return quantity is not None and (not isinstance(quantity, int) or quantity < 1)
 
 
-SpecificModelType = TypeVar("SpecificModelType", bound=Model)
+_M = TypeVar("_M", bound=Model)
+_NewM = TypeVar("_NewM", bound=Model)
```

Comment:
Forgive any lack of type annotation knowledge on my part (at least as it pertains to generics):

Why the need for two similar (seemingly identical) `TypeVars` here? Won't `_M` provide the sufficient use of a generic `Model` class throughout this module?

Also, I would prefer to see a more readable name for the new `TypeVar`. Suggestions are welcome, but I am thinking something along the lines of `GenericModel` (something a bit more descriptive). If the convention is to use a single letter wherever possible (like, say, Java), then I would prefer `M` without the leading underscore.

Reply:
A TypeVar represents the definition and reference to a type variable within the Generic. Because bakery has a `create` classmethod that generates a generic of a different type, we need a separate variable to represent this new definition, since it is independent of the current class. 

`flake8-pyi` introduces "Y001: Names of TypeVars in stubs should start with _." This makes sure you don’t accidentally expose names internal to the stub. It's just a habit that I've put them in like this even though it's not a stub. Happy to lose the leading underscore. From looking around, it's common-place for Python TypeVars to be just a single character, but if we want to be more explicit with the naming then I'm fine with that too.

Reply:
Yes, I think a single character would be fine (without the leading underscore).

And thank you for the information on the need for the second `TypeVar`.